src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -96,6 +114,10 @@ protected virtual void OnExpandPopulateNonFolders(IList<TreeNode> allChildren, T
             }
             foreach (var querier in queriers)
             {
+                if (querier.ValidFor != 0 && !querier.ValidFor.HasFlag(validForFlag))

```

Comment:
If you're doing this double-check in a bunch of places, it's a sign we should add a method on querier to see if it's valid for this version.